openpype/hosts/maya/api/lib.py

```diff
@@ -3446,3 +3447,47 @@ def dgcontext(mtime):
 def get_attribute_input(attr):
     connections = cmds.listConnections(attr, plugs=True, destination=False)
     return connections[0] if connections else None
+
+
+def get_color_management_preferences():
+    """Get and resolve OCIO preferences."""
+    data = {
+        # Is color management enabled.
+        "enabled": cmds.colorManagementPrefs(
+            query=True, cmEnabled=True
+        ),
+        "rendering_space": cmds.colorManagementPrefs(
+            query=True, renderingSpaceName=True
+        ),
+        "output_transform": cmds.colorManagementPrefs(
+            query=True, outputTransformName=True
+        ),
+        "output_transform_enabled": cmds.colorManagementPrefs(
+            query=True, outputTransformEnabled=True
+        ),
+        "view_transform": cmds.colorManagementPrefs(
+            query=True, viewTransformName=True
+        )
+    }
+
+    path = cmds.colorManagementPrefs(
+        query=True, configFilePath=True
+    )
+    # Resolve environment variables in config path. "MAYA_RESOURCES" are in the
+    # path by default.
+    for group in re.search(r'(<.*>)', path).groups():
+        path = path.replace(
+            group, os.environ[group[1:-1]]
+        )
```

Comment:
This breaks for me - `MAYA_RESOURCES` is not a defined environment variable on my end. The OCIO config does use it and also works fine in Maya, but I don't have it in my env vars.

Also watch out that the regex pattern here will match from the first `<` to the last `>` so will break on `<MAYA_RESOURCES>/hello/<SOME_OTHER_VAR>/world` since it matches `<MAYA_RESOURCES>/hello/<SOME_OTHER_VAR>` as one group.

Here's another way to substitute the groups:
```python
import re
import os

path = "<MAYA_RESOURCES>/OCIO-configs/Maya2022-default/config.ocio"

def subst_with_env_value(match):
    key = match.group(1)
    return os.environ.get(key, "")

path = re.sub(r'<([^>]+)>', subst_with_env_value, path)
print(path)
```
However due to the environment variable not existing for me that one still resolves to: `/OCIO-configs/Maya2022-default/config.ocio`

---

Unfortunately this also doesn't resolve the path:
```python
# Resolve environment variables in config path. "MAYA_RESOURCES" are in the
# path by default.

path = "<MAYA_RESOURCES%/OCIO-configs/Maya2022-default/config.ocio"

f = om.MFileObject()
f.setRawFullName(path)
print(f.resolvedFullName())
print(f.expandedFullName())
```

Reply:
Yeah, `MAYA_RESOURCES` is not an environment set when launching Maya natively so god knows why they included it in the default path for OCIO. We are account for the lacking environment variable here;
https://github.com/tokejepsen/pype/blob/enchancement/OP-2630_acescg_maya/openpype/hosts/maya/startup/userSetup.py#L7-L12

Good point about the regex. Didnt know that. Have replace it now.

Reply:
Actually - let's just remove it from userSetup.py - let's try to keep that as clean as possible and not influence the actual environment where we don't need to. I don't see why we need to precache something like that - it's not heavy at all to retrieve it? So let's avoid cluttering different locations of the code if we can.

However, I've found a way using the API to get the absolute path to resources.

I'd say for the logic in this method - let's do this:
```python
    path = cmds.colorManagementPrefs(
        query=True, configFilePath=True
    )

    # The OCIO config supports a custom <MAYA_RESOURCES> token
    maya_resources_token = "<MAYA_RESOURCES>"
    resources_path = om.MGlobal.getAbsolutePathToResources()
    path = path.replace(maya_resources_token, maya_resources_path)

    # Should we expand any remaining vars?
    path = os.path.expandvars(path)
```

That seems to be the only 'absolute path' getter on the `MGlobal` so I assume it's the only really special one?

For what it's worth - it seems that V-Ray also does this hardcoded replacement of this token in function `vrayGetMayaOCIOConfig` in `vray_maya_utils.py` . I think we should do, especially because other tokens, e.g. `<MAYA_LOCATION>` don't work even if that is an environment variable. It seems `<MAYA_RESOURCES>` is just special.

Reply:
Nice find!

Although what do you intent with using `path = os.path.expandvars(path)`?
Seems like we should need some logic to convert `<>` to OS native environment string variables.

Reply:
> Seems like we should need some logic to convert <> to OS native environment string variables.

For as far as I can see the `<MAYA_RESOURCES>` is the only token supported in the ocio config path, e.g. `<MAYA_LOCATION>` (which is in my `os.environ`) does not resolve - as such, these tokens aren't converted to OS environment variables to begin with. It seems `<MAYA_RESOURCES>` is just special - which can also be see by the fact that it's the only path retrievable from MGlobal in the API like that.

The expand vars was just to potentially replace any environment variables a user might have put in the path to resolve the actual full path. Not sure if needed - could also start without that in. It would've been useful if e.g. you'd start subprocesses with only a part of the environment variables needing to resolve the OCIO config path.

Reply:
As a new user with the default `<MAYA_RESOURCES>` in the path, I would expect to be able to use `<>` for environment variables:

```python
import maya.api.OpenMaya as om
import os


path = "<MAYA_RESOURCES>/hello/<SOME_OTHER_VAR>"
os.environ["SOME_OTHER_VAR"] = "buddy"


# The OCIO config supports a custom <MAYA_RESOURCES> token
maya_resources_token = "<MAYA_RESOURCES>"
maya_resources_path = om.MGlobal.getAbsolutePathToResources()
path = path.replace(maya_resources_token, maya_resources_path)

# Should we expand any remaining vars?
path = os.path.expandvars(path)
print(os.environ["SOME_OTHER_VAR"])
print(path)
```
But maybe we should just fix `<MAYA_RESOURCES>` for now and see what pitfalls users get into?

Reply:
> As a new user with the default <MAYA_RESOURCES> in the path, I would expect to be able to use <> for environment variables.

You might think that but it's not how Maya works - so I wouldn't try to implement that. If the path would've contained `<SOME_OTHER_VAR>` it wouldn't work out of the box in maya.